staging/src/k8s.io/apiserver/pkg/apis/auditregistration/validation/validation.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	audit "k8s.io/apiserver/pkg/apis/audit/validation"
+	auditconf "k8s.io/apiserver/pkg/apis/auditregistration"
+)
+
+// ValidateAuditSink validates the AuditSinks
+func ValidateAuditSink(as *auditconf.AuditSink) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, ValidateAuditSinkSpec(as.Spec)...)
```

Comment:
need to call ValidateObjectMeta with the name validation for this object. The one called generically by the store is a backstop with minimal name validation to protect against storage traversals